src/app/service/in-memory-metadata-file.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { Observable, of, BehaviorSubject } from 'rxjs';
+import { MetadataFileService, MetadataFile } from './metadata-file.service';
+
+// TODO
+export class InMemoryFile implements MetadataFile {
+  static counter = 0;
+  id: string;
+  name: string;
+
+  constructor(name: string) {
+    this.id = `${InMemoryFile.counter++}`;
+    this.name = name;
+  }
+}
+
+@Injectable()
+export class InMemoryMetadataFileService extends MetadataFileService {
+
+  files: BehaviorSubject<MetadataFile>[] = [];
+
+  listObs = new BehaviorSubject<MetadataFile[]>([]);
+  userLoggedInObs: Observable<boolean> = of(true);
+
+  constructor() {
+    super();
+  }
+
+  // @override
+  createFile(): Observable<MetadataFile> {
+    const file = new BehaviorSubject(new InMemoryFile('Untitled'));
+    this.files.push(file);
+    this.listObs.next(this.files.map((fileObs) => fileObs.value));
+    return file;
+  }
+
+  // @override
+  getFile(fileId: string): BehaviorSubject<MetadataFile>|null {
+    const file = this.files.find((fileObs) => fileObs.value.id === fileId);
+    if (file !== undefined) {
+      return file;
+    }
+    return null;
+  }
+
+  // @override
+  updateFile(file: MetadataFile): BehaviorSubject<MetadataFile>|null {
```

Comment:
Does the InMemory implementation of MetadataFile support persistence of the uxgraphs anywhere?

(sorry if I missed it reading through the PR..)